Gemfile

```diff
@@ -62,11 +65,12 @@ group :system_tests do
   gem "beaker-vmpooler", '~> 1.3'
   gem "serverspec", '~> 2.39'
   gem "beaker-pe",                                                               :require => false
-  gem "beaker-rspec", *location_for(ENV['BEAKER_RSPEC_VERSION'] || '~> 6.2')
+  gem "beaker-rspec", *location_for(ENV['BEAKER_RSPEC_VERSION'])
   gem "beaker-hostgenerator", *location_for(ENV['BEAKER_HOSTGENERATOR_VERSION'])
   gem "beaker-abs", *location_for(ENV['BEAKER_ABS_VERSION'] || '~> 0.1')
   gem 'pdk', *location_for(ENV['PDK_GEM_VERSION'])
-  gem "puppet-blacksmith", '~> 3.4',                                             :require => false
+  gem "puppet-blacksmith", '~> 3.4',                                             :require => false if Gem::Version.new(RUBY_VERSION.dup) < Gem::Version.new('7.0.0')
+  gem "puppet-blacksmith", '~> 6',                                               :require => false if Gem::Version.new(RUBY_VERSION.dup) >= Gem::Version.new('7.0.0')
```

Comment:
You probably wanted to check and compare the puppet version here?

Reply:
Great catch! Corrected.